Sources/TuistLoader/Models/FileListGlob+Unfold.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+import ProjectDescription
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+extension FileListGlob {
+    func unfold(
+        generatorPaths: GeneratorPaths,
+        filter: ((AbsolutePath) -> Bool)? = nil
+    ) throws -> [AbsolutePath] {
+        let resolvedPath = try generatorPaths.resolve(path: glob)
+        let resolvedExcluding = try resolvedExcluding(generatorPaths: generatorPaths)
+        let pattern = String(resolvedPath.pathString.dropFirst())
+        return FileHandler.shared.glob(AbsolutePath.root, glob: pattern).filter { path in
+            guard !resolvedExcluding.contains(path) else {
+                return false
+            }
+            if let filter = filter {
+                return filter(path)
+            }
+            return true
+        }
+    }
+
+    private func resolvedExcluding(generatorPaths: GeneratorPaths) throws -> Set<AbsolutePath> {
+        guard !excluding.isEmpty else { return [] }
+        var result: Set<AbsolutePath> = []
+        try excluding.forEach { path in
+            let resolved = try generatorPaths.resolve(path: path).pathString
+            let absolute = AbsolutePath(resolved)
+            let globs = AbsolutePath(absolute.dirname).glob(absolute.basename)
+            result.formUnion(globs)
```

Comment:
I believe `resolved` is already an `AbsolutePath` - there is no need to round trip from `String` > `AbsolutePath` again

```suggestion
            let resolved = try generatorPaths.resolve(path: path)
            let globs = AbsolutePath(resolved.dirname).glob(resolved.basename)
            result.formUnion(globs)
```

Reply:
updated